frontend/src/modules/editor/components/UnsavedChanges.js

```diff
@@ -0,0 +1,76 @@
+/* @flow */
+
+import * as React from 'react';
+import { Localized } from 'fluent-react';
+
+import './UnsavedChanges.css';
+
+
+type Props = {|
+    callback: ?Function,
+    hide: () => void,
+    ignore: () => void,
+    show: boolean,
+|};
+
+
+/*
+ * Renders the unsaved changes popup.
+ */
+export default class UnsavedChanges extends React.Component<Props> {
+    hideUnsavedChanges = () => {
+        this.props.hide();
+    }
+
+    leaveAnyway = () => {
+        const { callback, hide, ignore } = this.props;
+        if (callback) {
+            ignore();
+
+            setTimeout(() => {
+                callback();
+                hide();
+            }, 10);
```

Comment:
I wonder how to circumvent the `setTimeout` here. Without it, the callback (which depends on state) executes before the state is updated.

Reply:
I think you want to call the callback after the component's next render, in `componentDidUpdate`. In React the "correct" way to be sure that state has been updated is to wait for the next render to happen.